File: demo_dagger/src/main/java/com/mandria/android/mvp/example/MVPApplication.java
package com.mandria.android.mvp.example;

import android.app.Application;

import com.mandria.android.mvp.MVPLogger;
import com.mandria.android.mvp.example.di.components.AppComponent;
import com.mandria.android.mvp.example.di.components.DaggerAppComponent;


public class MVPApplication extends Application {

    public static AppComponent mAppComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        MVPLogger.SHOW_MVP_LOGS = true;

        initializeInjector();
    }

    /**
     * Initializes the injector.
     */
    private void initializeInjector() {
        mAppComponent = DaggerAppComponent
                .builder()
                .application(this)
                .build();
    }
}

File: demo_dagger_kotlin/src/main/kotlin/com/mandria/android/mvp/kotlin/example/MVPApplication.kt
package com.mandria.android.mvp.kotlin.example

import android.app.Application
import com.mandria.android.mvp.MVPLogger
import com.mandria.android.mvp.kotlin.example.di.AppComponent
import com.mandria.android.mvp.kotlin.example.di.DaggerAppComponent

class MVPApplication : Application() {

    override fun onCreate() {
        super.onCreate()

        MVPLogger.SHOW_MVP_LOGS = true

        appComponent = DaggerAppComponent.builder().application(this).build()
    }

    companion object {
        lateinit var appComponent: AppComponent
    }
}